### ISSUE TITLE
Guidance: Redirects

### ISSUE THREAD

**Issue by kristinoletmuskat** [department-of-veterans-affairs/va.gov-team#83498]
### Description
We need to improve our redirects process and documentation, and ensure that it is communicated and published in platform/engineering documentation on va, so that teams understand what they need to do and redirects are implemented accurately

Changes to the current process must address:

- [x] - Modified process to reference shift of responsibility off PW team and onto requesting team with emphasis of IA approval and validation, and PW PR review
- [x] - Updates to links across the modernized experience to ensure our links are accurate - this includes identifying links within React pages, core drupal pages, and facility pages - process details are TBD
- [ ] - Communications process to digital media partners and business line stakeholders informing them of the change and requesting they update the links within their TeamSite pages
- [ ] - Refined acceptance criteria and process documentation
- [x] - Communication of the redirect process 
- [x] - Always linking to the core/canonical URL, **not the /introduction page** or other sub-page - reference in link guidance and content style guide
- [x] - Solidify trailing slash guidance (we think we want the trailing slash, but we need to double-check with PW and platform engineering)
- [x] - Document that we do not do redirects on any documents (sometimes they will have to change the name back to what it was. It is a case by case basis)
- [x] - Guidance for how redirect destinations should be selected (this is low priority and primarily for our internal documentation, and can be done after the core work mentioned above)

## IA Support

```[tasklist]
### IA deliverables
- [x] Edit current process documentation to add learnings from recent redirect issues
- [x] Set up meeting with Megan and Randi, PW to clarify process
- [x] Set up meeting with CMS team to get their feedback on process
- [x] Update process documentation from those 2 meetings
- [ ] ~Finalize developer-facing content with Jill~  Separating this work from this ticket
- [x] Pass off to Mikki and Erin to review
- [x] Erin publishes in wherever it should be published
- [ ] Any other documentation is updated (IA spec link, redirect ticket itself etc)
- [ ] Announcement is made to teams
```

## Acceptance Criteria
- [ ] Process is published on va platform site
- [ ] Process documentation is shared with teams

**Comment by kristinoletmuskat** [comment]
**Notes from Mikki:**
- product teams should be able to do all redirects themselves for their tools AND drupal pages 
- public websites might still do some redirects (like education migration, family member hub, or CAIA projects)
- we need to rewrite the documentation (pw- engineering, caia- process) so that is explains how teams can do that. 
- every redirect will still come through IA
- PW will still do PR reviews
- CAIA should do the final validation in production

**Comment by kristinoletmuskat** [comment]
@erinrwhite @mnorthuis we have a few open questions to PW/CMS teams in this doc (that we plan to address next week), but if you would like to review the current draft, it's ready. Apart from some minor details, would be great to get your input on the overall placement and proposed structure for this content (as well as what we've proposed including for general consumption by teams).

**Comment by kristinoletmuskat** [comment]
Moved into blocked because waiting on feedback from PW, Erin, Mikki.

**Comment by mnorthuis** [comment]
@kristinoletmuskat I think the link to the doc is missing?

**Comment by kristinoletmuskat** [comment]
Whoops! Here is the doc: https://docs.google.com/document/d/1wRcgJn7p4Cuz4cFr-WF3JlvttK5CAbenZkJM4EqBLIs/edit#heading=h.efzncwe2ayez

**Comment by mnorthuis** [comment]
@kristinoletmuskat Left feedback and changes in the doc.  

I want to divide this guidance into 2 - part for VADS and part of internal use.  I made a note of that towards the beginning of the doc and highlighted what I want to be part of VADS documentation.  Once you have a chance to go through my comments and edits, and make any revisions, send it my way for review.   

I would like to focus on the VADS portion and then work with @erinrwhite to get that published.   For the rest, we will need to work with both PW and platform documentation to create a single SOT location.  I noticed they moved their documentation in GH, but I think this ultimately needs to go into the platform doc site with the engineering info.

**Comment by kristinoletmuskat** [comment]
Update: I went through the same [draft doc](https://docs.google.com/document/d/1wRcgJn7p4Cuz4cFr-WF3JlvttK5CAbenZkJM4EqBLIs/edit) and divided the guidance into to -- what's for VADS vs Platform site. I also migrated in developer-facing content from [PWs documentation](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md#4-page-level-cross-domain-redirects-from-subdomains-subdomainvagov-to-wwwvagov), and asked Jill to look through.

**Comment by jilladams** [comment]
I revised the table / Cases 1-8 section that was from the redirect strategy doc. I'm done in the Google doc, and didn't use track changes for those edits, so it's good to go from my perspective. (Tho I'm really hoping the Platform content people will help make it make sense!! It's so hard to write about / explain.)

**Comment by mnorthuis** [comment]
@kristinoletmuskat I made a few tweaks to the VADS section - am going to focus on that for now to get general guidance out there.  

For the platform side, we need to look at what exists on the platform site already and how @jilladams documentation fits in or replaces that.  Agree that we should work the platform content team on that.

**Comment by kristinoletmuskat** [comment]
Thanks @mnorthuis -- went through and made changes based on your comments. I remember you had said you asked the platform team for any content related to this -- do you have that list we could compare with Jill's documentation?

**Comment by mnorthuis** [comment]
They just did a search for "redirect" on the Platform site and sent me a link to the results...which was not helpful.  So, we can do the same and start to narrow down to pages that are related and need to be updated.   I can then engage someone from the engineering CoP to help us determine placement and how to handle the updates.

**Comment by erinrwhite** [comment]
Submitted [pull request to add this guidance](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/pull/3008)

Page content based on [GH markdown file](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/information-architecture/standards/redirect-standards.md)

**Comment by erinrwhite** [comment]
update 7/25:

- [Erin will update PR](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/pull/3008) to remove Platform website link then Mikki will review
- Mikki will work with Platform website crew on backend documentation 
- After platform documentation is updated, we'll add a link from VADS

**Comment by erinrwhite** [comment]
@mnorthuis [PR ready](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/pull/3008) for your review 👍🏻

**Comment by erinrwhite** [comment]
[Guidance published in the design system](https://design.va.gov/components/url-standards/redirects)! We need to announce this change, as well as our [updated title tag guidance](https://design.va.gov/content-style-guide/title-tags), to teams before we can close.

**Comment by mnorthuis** [comment]
Communication will go on my to do list. I will likely hold on it for a few days to see where the implementation conversations go. 

For the rest of the work here, we can remove reference to the implementation documentation, as that will be primarily handled by PW and platform once the security review and process decisions are complete.

**Comment by jilladams** [comment]
@kristinoletmuskat @mnorthuis FYI that Michelle has asked us to have an engineer take one more pass at the Redirect documentation, just to make sure we have no further changes in order to get other technical teams onboarded to making these changes. 

We are tracking that in #18782 . I've asked our team to flag in this ticket if they have any further edits, for your awareness, since you'll need to push any edits forward to the Content team.

**Comment by jilladams** [comment]
(If that Google doc is no longer the best place to review content, please let us know where to look instead)

**Comment by mnorthuis** [comment]
@jilladams there are 3 parts to the documentation in the Google doc:
1) IA guidance for when and why to use redirects - this is complete and has been published to the design system
2) Process and responsibilities info - this is drafted, but can't be finished until we know what the process and responsibilities are
3) Technical implementation documentation - this is a straight copy of your GH info and hasn't really been modified except to call out where we can crosslink.  

I'm removing 2 and 3 from this ticket given the conversation has changed a bit. 

#3 can be turned back over to you/your team, you can update your info back in your original GH page, and we will remove it from this doc so there aren't 2 versions out there
#2 we can come back to once we figure out who is doing what

**Comment by jilladams** [comment]
I did modify section 3 quite a bit from what was in the Github doc to make it more legible for folks long term. Is it possible to put that change back so I can copy it out to the Github doc for longevity, that in mind? 

Is the long-term idea now not to maintain the technical details in the Platform content site? If not, and if Public Websites no longer owns that information bc we don't own redirects, who will maintain it?

**Comment by jilladams** [comment]
Realized i can grab the content from the revision history, but I think clarifying the question around ownership / maintenance of the content for the technical process would still be valuable.

**Comment by mmiddaugh** [comment]
@jilladams the plan is still to house the technical details in the Platform Documentation Site and we will submit it for submission once we've completed our review. One it is handed off, PW will no longer be responsible for maintaining the documentation long term although the new owners are TBD.

**Comment by jilladams** [comment]
Ok thanks! I did go ahead and retrieve the edits from the Google doc, and port them back to Github, so https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md is now updated.

**Comment by mnorthuis** [comment]
Standards and guidance announced in slack, and process/responsibilities will be handled separately from this documentation issue, so I'm closing this!  Thank you all!
